### DISCUSSION TITLE
Spaces behave like tags or like folders depending on how you look at them

### DISCUSSION THREAD

**Discussion by davidegirardi** [element-hq/element-meta#2381] [Product Feedback]
I noticed that spaces behaves differently depending on how you look at them.

Let's take the following structure, names starting with a capital letter are spaces and the ones starting in small caps are rooms:

```
RootSpace/
├── r1
├── r2
└── ChildSpace
    ├── c1
    └── GrandChildSpace
        └── gc1
```

If I invite a user to RootSpace, they will see ChildSpace and the rooms r1 and r2.

If I invite the same user to GrandChildSpace, that space will not land directly in the sidebar when they accept. Same applies if I invite the user to gc1 or gc1, those rooms will not belong to RootSpace until I invite the same user to all the spaces.

At the same time the user will see ChildSpace and GrandChildSpace when checking RootSpace:
![image](https://github.com/element-hq/element-meta/assets/16451191/2252eb9a-2502-479a-a7eb-0987df9580fc)

This could make sense if spaces were like folders but they aren't. Once a user is invited to all the spaces (RootSpace, ChildSpace and GrandChildSpace) they behave like tags. The users see all the rooms under RootSpace (r1, r2, c1, gc1), many rooms (c1, gc1) in ChildSpaces etc...

Screenshot for a user member of all the spaces:
![image](https://github.com/element-hq/element-meta/assets/16451191/2fc23944-697e-4465-b028-062e945c4f41)

This creates some confusion since sometimes spaces are like tags and sometimes like folders: a room is member of a parent-space only if the user is also member of the hierarchy of sub-spaces leading to it.

**Comment by daniellekirkwood** [comment]
@gaelledel this will be interesting feedback for you ! :)

**Comment by TheArcaneBrony** [comment]
Just found out this discussion exists, this has been a major pain point for me with Element, as I use nested spaces a lot (leading up to hundreds of rooms if you recurse the tree). I do like the way Nheko handles this, in comparison (showing subspaces at the top of the room list, rather than recursing). I wonder if this may be a worthwhile change for Element?
